nuxeo-mongodb-directory/src/main/java/org/nuxeo/directory/mongodb/MongoDBSession.java

```diff
@@ -127,9 +132,19 @@ public DocumentModelList getEntries() throws DirectoryException {
     @Override
     public DocumentModel createEntry(Map<String, Object> fieldMap) throws DirectoryException {
         checkPermission(SecurityConstants.WRITE);
-        String id = String.valueOf(fieldMap.get(getIdField()));
-        if (hasEntry(id)) {
-            throw new DirectoryException(String.format("Entry with id %s already exists", id));
+        String id;
+        if (autoincrementId) {
+            UIDGeneratorService uidGeneratorService = Framework.getService(UIDGeneratorService.class);
+            UIDSequencer seq = uidGeneratorService.getSequencer();
+            Long longId = seq.getNextLong(SEQ_DIRECTORY);
```

Comment:
Here the code will share the sequence between all directories having an auto increment id, which is not what user will expect.

Also I'd rather not delegate to the `MongoDBUIDSequencer`. The code for auto increment management is simple enough that in can be done directly in the directory code. Also this way you can more easily use a collection name that's derived from the directory's collection name, by following standard MongoDB conventions (`mydir` -> `mydir.counters`).